pandas/core/dtypes/inference.py

```diff
@@ -263,7 +263,8 @@ def is_list_like(obj):
     """
 
     return (hasattr(obj, '__iter__') and
-            not isinstance(obj, string_and_binary_types))
+            not isinstance(obj, string_and_binary_types) and
+            isinstance(obj, collections.Iterable))
```

Comment:
import Iterable directly to the name space at the top

Reply:
done